aten/src/ATen/native/mps/operations/ReduceOps.mm

```diff
@@ -1959,14 +1991,23 @@ Tensor median_mps(const Tensor& input_t) {
                                                         length:1
                                                         name:nil];
               MPSGraphTensor* argreduceOutTensor = nil;
-                argreduceOutTensor = [mpsGraph argSortWithTensor:inputTensor
+                argreduceOutTensor = [mpsGraph argSortWithTensor:castInputTensor
                                                                         axis:(NSInteger)dim_
                                                                         name:@"argmax_out"];
               MPSGraphTensor* argOutputTensor = [mpsGraph sliceTensor:argreduceOutTensor
                                                         dimension:dim_
                                                         start:((NSUInteger) (int)((dim_total_elements+1)/2 ) - 1)
                                                         length:1
                                                         name:nil];
+              // Cast back
+              dataType = native_mps::getMPSDataType(input_t.scalar_type());
+              if (dataType != MPSDataTypeInt32 &&
+                  dataType != MPSDataTypeFloat32 &&
+                  dataType != MPSDataTypeFloat16) {
+                  argreduceOutTensor = [mpsGraph castTensor:argreduceOutTensor
```

Comment:
You would also need to cast back also the `outputTensor` back to the original dtype of input (if input was casted)